Code/Framework/AzCore/AzCore/DOM/DomPath.cpp

```diff
@@ -0,0 +1,455 @@
+                    {
+                        ++size;
+                    }
+                    ++size;
+                }
+            }
+        }
+        return size;
+    }
+
+    void Path::FormatString(char* stringBuffer, size_t bufferSize) const
+    {
+        size_t bufferIndex = 0;
+
+        auto putChar = [&](char c)
+        {
+            if (bufferIndex == bufferSize)
+            {
+                return;
+            }
+            stringBuffer[bufferIndex++] = c;
+        };
+
+        auto writeToBuffer = [&](const char* key)
+        {
+            for (size_t keyIndex = 0; key[keyIndex]; ++keyIndex)
+            {
+                const char c = key[keyIndex];
+                if (c == EscapeCharacter)
+                {
+                    putChar(EscapeCharacter);
+                    putChar(TildeSequence);
+                }
+                else if (c == PathSeparator)
+                {
+                    putChar(EscapeCharacter);
+                    putChar(ForwardSlashSequence);
+                }
+                else
+                {
+                    putChar(c);
+                }
+            }
+        };
+
+        for (const PathEntry& entry : m_entries)
+        {
+            putChar(PathSeparator);
+            if (entry.IsEndOfArray())
+            {
+                putChar(EndOfArrayCharacter);
+            }
+            else if (entry.IsIndex())
+            {
+                bufferIndex += azsnprintf(&stringBuffer[bufferIndex], bufferSize - bufferIndex, "%zu", entry.GetIndex());
+            }
+            else
+            {
+                writeToBuffer(entry.GetKey().GetCStr());
+            }
+        }
+
+        putChar('\0');
+    }
+
+    AZStd::string Path::ToString() const
+    {
+        AZStd::string formattedString;
+        const size_t size = GetStringLength();
+        formattedString.resize_no_construct(size);
+        FormatString(formattedString.data(), size + 1);
+        return formattedString;
+    }
+
+    void Path::FromString(AZStd::string_view pathString)
+    {
+        m_entries.clear();
+        size_t pathIndex = 0;
+        bool isNumber = true;
+        for (size_t i = 0; i <= pathString.size(); ++i)
+        {
+            if (i == pathString.size() || pathString[i] == PathSeparator)
+            {
+                if (i > 0)
+                {
+                    AZStd::string_view section(&pathString.data()[pathIndex], i - pathIndex);
+                    if (section.size() == 1 && section[0] == EndOfArrayCharacter)
+                    {
+                        PathEntry entry;
+                        entry.SetEndOfArray();
+                        m_entries.push_back(AZStd::move(entry));
+                    }
+                    else if (isNumber && section.size() > 0)
+                    {
+                        size_t index = 0;
+                        ConsoleTypeHelpers::StringToValue(index, section);
```

Comment:
If you used `strtoull` that would allow you to update a char* pointer that points the character right after the number has been converted.

Reply:
Yeah that's handy, but not needed here because we already know the section length in advance.